lib/test.js

```diff
@@ -239,31 +233,45 @@ Test.prototype._publicApi = function () {
 		self._assert();
 	}
 
+	function onAssertionEvent(event) {
+		if (event.type === 'error') {
+			event.error.powerAssertContext = event.powerAssertContext;
+			event.error.originalMessage = event.originalMessage;
+			self._setAssertError(event.error);
+			self._assert();
+			return null;
+		}
+
+		var fn = observableToPromise(event.returnValue);
+
+		if (isPromise(fn)) {
+			return Promise.resolve(fn)
+				.catch(function (err) {
+					self._setAssertError(err);
```

Comment:
Given your answer below I understood what is `onAssertionEvent` for, but I still don't understand what would the answer to my question be.

Here, if assertion returns promise and that promise rejects, we just `_setAssertError()`, but without `powerAssertContext` like above (lines 237-242).

Or (correct me if I'm wrong) if "in that promise" there is a failed assertion, `onAssertionEvent` will be triggered again with `event.type === 'error'`. In that case, we don't need a `.catch()` block on that promise, don't we?